reconcile/external_resources/integration_secrets_sync.py

```diff
@@ -17,16 +24,24 @@ def run(dry_run: bool, thread_pool_size: int) -> None:
     the target clusters
     """
     vault_settings = get_app_interface_vault_settings()
-    secrets_reader = create_secret_reader(use_vault=vault_settings.vault)
+    secret_reader = create_secret_reader(use_vault=vault_settings.vault)
     er_settings = get_settings()[0]
+
     namespaces = [ns for ns in get_namespaces() if ns.external_resources]
     er_inventory = ExternalResourcesInventory(namespaces)
+    m_inventory = load_module_inventory(get_modules())
+
+    to_sync_specs = []
+    for key, spec in er_inventory.items():
+        module = m_inventory.get_from_external_resource_key(key)
+        if module.outputs_secret_sync:
+            to_sync_specs.append(spec)
```

Comment:
To me, it's harder to read. I don't see any advantage of writing it this way honestly.

Reply:
oh, sorry, I thought `module` is appended, then just

```python
    to_sync_specs = [
        spec
        for key, spec in er_inventory.items()
        if m_inventory.get_from_external_resource_key(key).outputs_secret_sync
    ]
```

if get `to_sync_specs` is extracted in a method with some name like `build_to_sync_specs`, then it doesn't matter which syntax to use as we can tell the intention from method name. When inline implementation, it will take a while to understand what these `for` loop and `if` condition are doing, what's the final thing needed later. Using list comprehension makes it easier to understand, as in some other languages, writing `xxx = []` then mutating it in loop will show warnings and auto convert to functional `map` `filter` equivalent code.

Reply:
Added.